pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -248,6 +248,26 @@ func (uc *UpgradeCluster) upgradable(currentVersion string) error {
 	return errors.Errorf("%s cannot be upgraded to %s", currentVersion, targetVersion)
 }
 
+// getVMSSNodeVersion returns a VMSS node's current Kubernetes version via Kubernetes API.
+// We can't rely on tags because VMSS does not support tags for individual instances.
+func (uc *UpgradeCluster) getVMSSNodeVersion(client armhelpers.KubernetesClient, name string, tags map[string]*string, latestModelApplied bool) string {
+	if latestModelApplied && tags != nil && tags["orchestrator"] != nil {
+		parts := strings.Split(*tags["orchestrator"], ":")
+		if len(parts) == 2 {
+			return parts[1]
```

Comment:
this seems to me like it does rely on the tag as the first attempt to get the k8s version which seems to be the opposite of what the PR description and function comment indicate

Reply:
updated the comments and the PR title.